File: cpp/198houseRobber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        int m = nums[0];
        int temp = 0;
        int temp2 = 0;
        for(int i = 1 ; i < n; i++){
            temp2 = m;
            m = max(m, nums[i] + temp);
            temp = temp2;
        }
        return m;
    }
};

File: python/198HouseRobber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        r = 0
        first = 0
        second = nums[0]
        for i in range(1,len(nums)):
            third = max(first+nums[i],second)
            first = second
            second = third
        return second